### ISSUE TITLE
clickhouse-jdbc: the count of view less then actual

### ISSUE THREAD

**Issue by justwinter** [ClickHouse/clickhouse-jdbc#1013]
Hi,everybody:
When I used clickhouse-jdbc driver to querry clickhouse view ,the cout result less then actual count.

Details:
I have a distributed table in the clickhouse ,and create view based on the distributed table .
When the count of distributed table is small,the query is normal. At this time, the count of the view is the same as that of the distribution table.

However, when the count of the distributed table is large, for example, when the count is greater than 1000000 rows, the query is abnormal. At this time, the count of views is different from that of distributed tables, only a little more than that of local tables.

The problem occurs when I use jdbc driver to query Clickhouse view, but when I use command-line client: clickhouse-client or curl command to query, the count of the view is normal. I have tried all JDBC versions below 0.3.2 and got the same results .

My clickhouse version is 22.1.3.7

Have you encountered similar problems? Or do you have any good solutions?

Here is my test SQL：
CREATE TABLE IF NOT EXISTS temp.test_table_local ON CLUSTER cluster_test
(
id String
,name String
,desc String
,w_insert_dt String
) ENGINE = ReplacingMergeTree ORDER BY id SETTINGS index_granularity = 8192;

CREATE TABLE IF NOT EXISTS temp.test_table ON CLUSTER cluster_test
(
id String
,name String
,desc String
,w_insert_dt String
) ENGINE = Distributed(cluster_test,temp,test_table_local,metroHash64(id));

create view temp.test_table_v ON CLUSTER cluster_test as
select * from temp.test_table;

insert into temp.test_table
select concat(toString(t.uuid),toString(b.sort_num)),t.name,t.comment,now()
from system.tables t
left join (select arrayJoin( range(1,1000)) as sort_num ) b on 1=1
where t.uuid not in ('00000000-0000-0000-0000-000000000000');

select count(1),max(id) from temp.test_table;
-- 2081916 fff61eea-90d4-471d-bff6-1eea90d4471d999

select count(1),max(id) from temp.test_table_v;
-- 742035 fff61eea-90d4-471d-bff6-1eea90d4471d995

select count(1),max(id) from temp.test_table_local;
-- 692883 fff61eea-90d4-471d-bff6-1eea90d4471d995

**Comment by zhicwu** [comment]
Hi @justwinter, I noticed the same issue in server repo - @den-crane suggested you to try curl. Did you get same result using curl? Just trying to understand if the issue is related to http interface.

**Comment by justwinter** [comment]
> Hi @justwinter, I noticed the same issue in server repo - @den-crane suggested you to try curl. Did you get same result using curl? Just trying to understand if the issue is related to http interface.

Yes, I tried the curl command, and the query result is normal. Only when I use JDBC to query, it is abnormal

**Comment by zhicwu** [comment]
> Yes, I tried the curl command, and the query result is normal. Only when I use JDBC to query, it is abnormal

What's the IDE you're using? DBeaver or a program using the JDBC driver? Is there any proxy between client and server? It sounds like an issue since curl worked just fine. I'll follow the instructions you provided and see if I can reproduce.

**Comment by justwinter** [comment]
> What's the IDE you're using? DBeaver or a program using the JDBC driver? Is there any proxy between client and server? It sounds like an issue since curl worked just fine. I'll follow the instructions you provided and see if I can reproduce.

I'm using DBeaver 21.1.2，and no agent or proxy between client and server

**Comment by den-crane** [comment]
>or curl command to query, the count of the view is normal. I have tried all JDBC versions below 0.3.2 and got the same results .

Clarify he same results. If use JDBC versions below 0.3.2 is it correct or incorrect result?

**Comment by justwinter** [comment]
> Clarify the same results. If you use JDBC versions below 0.3.2 is it correct or incorrect result?

incorrect result

**Comment by den-crane** [comment]
What if you query *SELECT toUInt64(9876543) *what result do you get?

**Comment by justwinter** [comment]
> What if you query *SELECT toUInt64(9876543) *what result do you get?

Get the correct value 9876543

**Comment by justwinter** [comment]
My cluster has three machines. When I connect to different IP addresses, the query results will be different.
Let's see

IP 1:
select count(1),max(id) from temp.test_table;
-- 2081916	fff61eea-90d4-471d-bff6-1eea90d4471d999

select count(1),max(id) from temp.test_table_v;
-- 742709	fff61eea-90d4-471d-bff6-1eea90d4471d999

select count(1),max(id) from temp.test_table_local;
-- 693557	fff61eea-90d4-471d-bff6-1eea90d4471d999

IP 2:
select count(1),max(id) from temp.test_table;
-- 2081916	fff61eea-90d4-471d-bff6-1eea90d4471d999

select count(1),max(id) from temp.test_table_v;
-- 742709	fff61eea-90d4-471d-bff6-1eea90d4471d999

select count(1),max(id) from temp.test_table_local;
-- 693557	fff61eea-90d4-471d-bff6-1eea90d4471d999

IP 3:
select count(1),max(id) from temp.test_table;
-- 2081916	fff61eea-90d4-471d-bff6-1eea90d4471d999

select count(1),max(id) from temp.test_table_v;
-- 742035	fff61eea-90d4-471d-bff6-1eea90d4471d995

select count(1),max(id) from temp.test_table_local;
-- 692883	fff61eea-90d4-471d-bff6-1eea90d4471d995

**Comment by den-crane** [comment]
@justwinter please share 

```
select name, value from system.settings where changed
```
executed via jdbc and curl

**Comment by zhicwu** [comment]
I didn't reproduce the issue on my 4-node(2 shards + 2 replicas + 3 keepers) cluster in local. I tried different database clients as well. Will try it again using 3-node without replica and generate different rows in local tables.

**Comment by justwinter** [comment]
> 
JDBC:
load_balancing	random
distributed_ddl_task_timeout	1200
max_bytes_before_external_group_by	8053063680
max_bytes_before_external_sort	8053063680
max_result_rows	200
result_overflow_mode	break
max_memory_usage	42949672960
allow_experimental_database_materialized_mysql	1

CURL:
max_threads	20
load_balancing	random
distributed_ddl_task_timeout	1200
max_bytes_before_external_group_by	8053063680
max_bytes_before_external_sort	8053063680
max_memory_usage	102400000000
allow_experimental_database_materialized_mysql	1

**Comment by justwinter** [comment]
@den-crane @zhicwu  I found max_result_rows/result_overflow_mode  these two settings are set by dbeaver, and I can't changes it.
 So I tried to upgrade dbeaver, and then query was normal.
I get the same and correct results when querying views and distributed tables.

 In addition, for Java programs using JDBC connection, setting a larger value for max_result_rows  in the JDBC string ,I get the correct value .

I'm very sorry for wasting your time because of this small problem. At the same time, thank you very much for your help!
